### PR TITLE
1228 port contextual menu pattern

### LINKED ISSUE #1228: Combine dashboard theme contextual-menu pattern

### PR DESCRIPTION
**Pull request by richmccartney**
## Done

- Ports over the contextual menu pattern and JS.

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- View the pattern by clicking on links at http://0.0.0.0:8101/vanilla-framework/examples/patterns/contextual-menu/contextual-menu/

## Details

Fixes #1228

### CODE REVIEW

File: docs/en/patterns/contextual-menu.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+collection: patterns
+title: Contextual menu
+---
+
+## Contextual menu
+
+A contextual menu can be applied to any button, link or navigation item that requires a secondary menu.
```
Comment (anthonydillon) [inline docs/en/patterns/contextual-menu.md]: Please add JavaScript documentation. With details on the hooks and how it works.

File: examples/patterns/contextual-menu/contextual-menu.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
+---
+layout: default
+title: Contextual menu
+category: _patterns
+---
+
+<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Modi natus <span class="p-contextual-menu">
+    <a href="#" class="u-toggle" aria-controls="#menu" aria-expanded="false" aria-haspopup="true">Link</a>
```
Comment (anthonydillon) [inline examples/patterns/contextual-menu/contextual-menu.html]: `u-toggle`should be part of this pattern. So you can use this pattern with out importing utilities.

File: docs/en/patterns/contextual-menu.md (hunk 1 above)
Comment (anthonydillon) [inline docs/en/patterns/contextual-menu.md]: Also mention that the menu id should be unique.

File: examples/patterns/contextual-menu/contextual-menu.html [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        for(i = 0; i < t.length; i++) {
+
+            // Set click event to each toggle control element
+            t[i].addEventListener('click', function(event) {
+
+                // Get toggle aria-control target
+                var target = document.querySelector(
+                    this.getAttribute('aria-controls'));
+
+                // If target hidden is already true set to fale and toggle
```
Comment (anthonydillon) [inline examples/patterns/contextual-menu/contextual-menu.html]: False

File: scss/_patterns_contextual-menu.scss [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      white-space: nowrap;
+      width: 100%;
+
+      &:hover {
+        background-color: $color-light;
+        text-decoration: none;
+      }
+    }
+
+    // Alignment extension to align the menu to the left
```
Comment (anthonydillon) [inline scss/_patterns_contextual-menu.scss]: These should not be extensions be a single class modified. Therefore, `--left` should extend the core pattern and apply it's modifications. Then its usage is a single class.

File: scss/_patterns_contextual-menu.scss [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        left: 0;
+
+        &::before,
+        &::after {
+          left: $sp-small;
+          right: initial;
+        }
+
+        &::after {
+          left: $sp-small + .1rem;
```
Comment (anthonydillon) [inline scss/_patterns_contextual-menu.scss]: Could use `$sp-medium` here instead of `1rem`.

Comment (richmccartney) [inline scss/_patterns_contextual-menu.scss]: I don't think there is a size that matches what design wanted hence why I did the said value.

Comment (anthonydillon) [inline scss/_patterns_contextual-menu.scss]: I mean `$sp-small + $sp-medium` which would be the same as you have.

File: examples/patterns/switch.html [hunk 6]
```diff
@@ -0,0 +1,117 @@
+---
+layout: default
+title: Switch
+category: _patterns
+---
+
```
Comment (anthonydillon) [inline examples/patterns/switch.html]: Shouldnt this switch pattern be apart of another PR?

File: examples/patterns/contextual-menu/contextual-menu.html [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+      </span>
+      <span class="p-contextual-menu__group">
+          <a href="#" class="p-contextual-menu__link">Link</a>
+          <a href="#" class="p-contextual-menu__link">Link</a>
+          <a href="#" class="p-contextual-menu__link">Link</a>
+      </span>
+  </span>
+</span>, consectetur adipisicing elit. Modi natus atque eligendi deleniti hic, dolores veritatis reiciendis officiis illo, facere facilis accusamus similique nulla nesciunt. Nemo reprehenderit officia assumenda error enim. Recusandae reiciendis ipsum, mollitia illo iusto excepturi alias dolore fugit eligendi nostrum, unde architecto consequuntur similique quo! Maxime iusto facere, commodi iste fuga officiis.</p>
+
+<h2>Example JavaScript</h2>
```
Comment (anthonydillon) [inline examples/patterns/contextual-menu/contextual-menu.html]: Dont need this javascript section as example js will show whats in the DOM.

File: scss/_patterns_contextual-menu.scss (hunk 5 above)
Comment (anthonydillon) [inline scss/_patterns_contextual-menu.scss]: Oh I see its .1rem. My fauilt.

**Comment by richmccartney** [comment]
@anthonydillon should all be done now.
